lib/ContractInteract/TokenRules.js

```diff
@@ -0,0 +1,180 @@
+   * @param {String } eip20Token EIP20 token contract address deployed for an economy.
+   * @param {Object} txOptions Tx options.
+   *
+   * @returns {Promise<TokenRules>} Promise containing the TokenRules
+   *                              instance that has been deployed.
+   */
+  static async deploy(auxiliaryWeb3, organization, eip20Token, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+
+    const tx = TokenRules.deployRawTx(auxiliaryWeb3, organization, eip20Token);
+
+    return Utils.sendTransaction(tx, txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new TokenRules(auxiliaryWeb3, address);
+    });
+  }
+
+  /**
+   * Method which returns Tx object to deploy TokenRules contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  static deployRawTx(auxiliaryWeb3, organization, eip20Token) {
+    if (!(auxiliaryWeb3 instanceof Web3)) {
+      throw new TypeError(`Mandatory Parameter 'auxiliaryWeb3' is missing or invalid: ${auxiliaryWeb3}`);
+    }
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const args = [organization, eip20Token];
+    const contract = Contracts.getTokenRules(auxiliaryWeb3, organization, eip20Token);
+
+    return contract.deploy({
+      data: bin,
+      arguments: args
+    });
+  }
+
+  /**
+   * It is used to register a custom rule to the economy.
+   *
+   * @param {String} ruleName Name of the rule.
+   * @param {String} ruleAddress Contract address of the rule.
+   * @param {String} ruleAbi Abi of the rule.
+   * @param {Object} txOptions Tx options.
+   *
+   * @returns {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async registerRule(ruleName, ruleAddress, ruleAbi, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+    const txObject = await this.registerRuleRawTx(ruleName, ruleAddress, ruleAbi);
+    return Utils.sendTransaction(txObject, txOptions);
+  }
+
+  /**
+   * Private method which is used to register a custom rule in the economy.
+   *
+   * @param {String} ruleName Name of the rule.
+   * @param {String} ruleAddress Contract address of the rule.
+   * @param {String} ruleAbi Abi of the rule.
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  registerRuleRawTx(ruleName, ruleAddress, ruleAbi) {
```

Comment:
Unit tests missing.

Reply:
It's there. Please check file: test/unit/TokenRules/registerRuleRawTx.js